substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageBFDNameProvider.java

```diff
@@ -375,22 +412,113 @@ public String bfdMangle(Member m) {
      * Make the provider aware of the current native libraries. This is needed because the provider
      * is created in a feature after registration but the native libraries are only available before
      * analysis.
-     * 
+     *
      * @param nativeLibs the current native libraries singleton.
      */
     public void setNativeLibs(NativeLibraries nativeLibs) {
         this.nativeLibs = nativeLibs;
     }
 
     private static class BFDMangler {
+
+        /*
+         * The mangler tracks certain elements as they are inserted into the mangled name and bind
```

Comment:
Excellent explanation. Thanks for adding that!

Reply:
You are welcome. Of course, I'll be thanking myself in 6 months time when I come to review this!